kafka-commons/src/main/java/uk/gov/homeoffice/digital/sas/kafka/listener/KafkaEntityListener.java

```diff
@@ -1,32 +1,37 @@
 package uk.gov.homeoffice.digital.sas.kafka.listener;
 
+import jakarta.persistence.PostPersist;
+import jakarta.persistence.PostRemove;
+import jakarta.persistence.PostUpdate;
 import jakarta.validation.constraints.NotNull;
 import uk.gov.homeoffice.digital.sas.kafka.message.KafkaAction;
+import uk.gov.homeoffice.digital.sas.kafka.message.Messageable;
 import uk.gov.homeoffice.digital.sas.kafka.producer.KafkaProducerService;
 
-public abstract class KafkaEntityListener<T> {
+public class KafkaEntityListener<T extends Messageable> {
 
   private final KafkaProducerService<T> kafkaProducerService;
 
   protected KafkaEntityListener(KafkaProducerService<T> kafkaProducerService) {
     this.kafkaProducerService = kafkaProducerService;
   }
 
-  public abstract String resolveMessageKey(T resource);
-
+  @PostPersist
```

Comment:
Do we want to keep this as post and not change it to pre?

Reply:
@Pre works in tandem with the Transaction Management synchronisation. So we should change it to the @Pre annotation only once the transaction management synchronisation piece is added to kafka-commons

Reply:
That makes sense to me.